### PR TITLE
Create docs for adding the service name field to logs

### LINKED ISSUE #4037: [Request] Document how to add `service.name` to logs
### Summary

**All details are in https://github.com/elastic/observability-docs/issues/4027 and https://github.com/elastic/observability-docs/issues/4027#issuecomment-2183019089.**

> We are delivering our first step in a new Observability Experience which will focus on showing users entities (e.g. Hosts, Containers, Services, K8s deployments etc)...even if they only have logs for them.
> 
> Our first step will be an updated Services Inventory which will show not only services instrumented with APM but also services we found in logs (reliant on service.name being declared in the data).
> 

> I just wanted to call out that I assume we'll need some pretty heavy expertise from our side in how users can add `service.name` to existing logs using things like:
> 
> - [Mapping](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping.html)
> - [Logstash](https://www.elastic.co/guide/en/logstash/current/pipeline.html#_outputs)
> - Processors : [filebeat](https://www.elastic.co/guide/en/beats/filebeat/current/defining-processors.html) | [elastic agent](https://www.elastic.co/guide/en/fleet/current/elastic-agent-processor-configuration.html)

### PR DESCRIPTION
**Pull request by mdbirnstiehl**
This PR closes #4037 and creates the docs that show users how to add the `service.name` field to their logs.

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-c2g5630i0-elastic-dev.vercel.app

* Built with commit 72e8f6d367de2232c06c82ae98b7d891773c8240

> Issues? Visit #next-docs in Slack

File: docs/en/serverless/logging/add-logs-service-name.mdx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+For example, adding the `add_fields` processor to the inputs section of a standalone ((agent)) or ((filebeat)) configuration would add `your_service_name` as the `service.name` field:
+
+```console
+processors:
+    - add_fields:
+        target: service
+        fields:
+            name: your_service_name
+```
```
Comment (mdbirnstiehl) [inline docs/en/serverless/logging/add-logs-service-name.mdx:29]: This is the way the logs onboarding adds the service name, not sure if this is the preferred way but used it for this first draft.

**Comment by mdbirnstiehl** [comment]
@roshan-elastic I've created this first draft to work from, please let me know of any feedback you might have.

**Comment by roshan-elastic** [comment]
Cheers @mdbirnstiehl - looking good!

A few questions:

- I'm trying to figure out how to do the alias field type...it looks like you've posted something to run in DevTools? Maybe this could be calIed out? I just want to try and have a 'hello world' which doesn't require users to read too many more docs to get it working if possible.

- I can also see there are different ways of handling mapping. For example:

_Stack Management > Index Management > Templates > Edit template_

image

Are there any existing docs which could point to the different ways to handle mapping?

- Is there a way to point users to things like ingest pipelines and logstash? I think making users aware that they can do these kinds of things will be pretty valuable

**Comment by mdbirnstiehl** [comment]
> * I'm trying to figure out how to do the alias field type...it looks like you've posted something to run in DevTools? Maybe this could be calIed out? I just want to try and have a 'hello world' which doesn't require users to read too many more docs to get it working if possible.
> * I can also see there are different ways of handling mapping. For example:
> 
> _Stack Management > Index Management > Templates > Edit template_
> 

> Are there any existing docs which could point to the different ways to handle mapping?

Yeah, maybe for this situation it would be easier for users to use the the UI in this example, and send them to the docs for other ways to update mapping. We have some [docs on index management](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-mgmt.html) that provide some insight into how you might create/update mapping. I'll take another look and see if we have anything that touches on using dev tools for updates.

> * Is there a way to point users to things like ingest pipelines and logstash? I think making users aware that they can do these kinds of things will be pretty valuable

I'll add a section of other scenarios/helpful links (i'll think of a title) to point users to these more complex use cases.

**Comment by mdbirnstiehl** [comment]
@roshan-elastic I've made the suggested updates if you want to take another look.

**Review by roshan-elastic — COMMENTED** [review]
Hey @mdbirnstiehl,

Looking good!

> Adding a service name also associates your logs with a service in the service inventory.

Is it worth mentioning that this will only show if you enable the new experience? It'll be opt-in for a few months at least

**Review by dedemorton — COMMENTED** [review]
Just a couple of minor things I noticed when I was scanning the new content.

**Comment by mdbirnstiehl** [comment]
@roshan-elastic Everything should be up-to-date with your comments if you want to give it one more look. I think we probably want a developer review as well before wrapping up the PR.

**Review by klacabane — APPROVED** [review]

**Comment by klacabane** [comment]
LGTM!

**Comment by smith** [comment]
> @smith - Wondering if you might have a suggestion of an engineer who might be able to review this short doc on declaring service.name (we won't block getting this out though)?

@roshan-elastic I asked @elastic/obs-ux-onboarding-team, since we want to make sure it's consistent for what we do with the logs onboarding flow.

File: docs/en/serverless/logging/add-logs-service-name.mdx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+Adding the `add_fields` processor to an integration settings would add `your_service_name` as the `service.name` field:
+
+<DocImage size="2" url="../images/add-field-processor.png" alt="Add the add_fields processor to an integration" />
+
+For more on defining processors, refer to [define processors](((fleet-guide))/elastic-agent-processor-configuration.html).
+
+## Map an existing field to the service name field
+
+For logs that already have the field to use as the service name, map that field to the `service.name` field using the [alias field type](((ref))/field-alias.html).
```
[the remark this answers is not in the document]
Comment (roshan-elastic) [reply docs/en/serverless/logging/add-logs-service-name.mdx:39]: Thanks @smith - LGTM!

**Review by dedemorton — APPROVED** [review]
LGTM

**Pull request merged**